server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/TraceServiceElasticsearch.java

```diff
@@ -314,4 +347,84 @@ public void clear(String tenantId) {
         }
     }
 
-}
+    public static class TraceSerializer extends JsonSerializer<Trace> {
+
+        /* (non-Javadoc)
+         * @see com.fasterxml.jackson.databind.JsonSerializer#serialize(java.lang.Object,
+         *          com.fasterxml.jackson.core.JsonGenerator, com.fasterxml.jackson.databind.SerializerProvider)
+         */
+        @Override
+        public void serialize(Trace trace, JsonGenerator jgen, SerializerProvider provider)
+                throws IOException, JsonProcessingException {
+            jgen.writeStartObject();
+            jgen.writeStringField(BUSINESS_TRANSACTION_FIELD, trace.getBusinessTransaction());
+            jgen.writeStringField(HOST_ADDRESS_FIELD, trace.getHostAddress());
+            jgen.writeStringField(HOST_NAME_FIELD, trace.getHostName());
+            jgen.writeStringField(ID_FIELD, trace.getId());
+            jgen.writeStringField(PRINCIPAL_FIELD, trace.getPrincipal());
+            jgen.writeNumberField(START_TIME_FIELD, trace.getStartTime());
+            jgen.writeArrayFieldStart(NODES_FIELD);
+            for (Node n : trace.getNodes()) {
```

Comment:
Tried it, and also trace.getNodes().forEach(n -> jgen.writeObject(n));, but geting "Unhandled exception type: IOException" in Eclipse (and also maven from command line) - even when a try/catch(Throwable) block surrounds.

Any ideas? Using jackson 2.6.0.